demos/yielding/README.md

```diff
@@ -0,0 +1,112 @@
+# Yielding
+
+In this demo, we try to show the usage of yielding, which is one characteristic of a multitasking system. 
+
+By calling syscall `yield()`, the thread gives away the utilization of a CPU to another thread, or a specific thread (will be discussed later in another demo).
+
+To achieve thread yielding, we need to introduce the following functions:
+
+* `add_thread_from_worker` (in `src/user/thread.c`)
+* `do_scheduler` (in `src/kernel/thread.c`)
+* `yield` (in `src/user/thread.c`)
+
+## `add_thread_from_worker`
+
+In AMT, we try to maintain all of the threads in an array called `all_threads`.
+
+By calling this function, the AMT will create a new thread instance with the given arguments, initialize the permissions, and put the function in. 
+
+After the creation of the thread, the entity will be put into `all_threads` array and wait for the utilization of the CPU.
```

Comment:
So since the thread array is statically allocated it's kinda already in the array. I think you can just leave that bit out completely.

"After the creation of the thread it will wait to be chosen by the scheduler so it can utilize the CPU"?

(also good to see "the cpu" here, so the "a cpu" above should definitely be changed)